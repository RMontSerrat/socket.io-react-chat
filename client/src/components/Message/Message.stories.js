import React from 'react';
import { storiesOf } from '@storybook/react';
import Message from './Message';

storiesOf('Message', module)
  .add('default', () => (
    <div style={{ padding: '20px' }}>
      <Message userName="Romulo" date={new Date()}>Fala galera</Message>
    </div>
  ))
  .add('without userName', () => (
    <div style={{ padding: '20px' }}>
      <Message userName="Servidor" date={new Date()}>Romulo entrou na sala...</Message>
    </div>
  ));

