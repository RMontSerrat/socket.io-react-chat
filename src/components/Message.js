import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  color: ${props => props.theme.colorPrimary};
  font-family: ${props => props.theme.fontFamily};
  padding: 5px;
  background-color: ${props => props.active ? props.theme.colorLightGreen : 'transparent'};
`

const DateContainer = styled.span`
  display: block;
  padding-right: 10px;
`
const Name = styled.strong`
  padding-right: 5px;
`

const Message = ({ children, userName = 'Servidor', date, active }) =>
  <Container active={active}>
    <DateContainer>[{new Date(date).toLocaleTimeString()}]</DateContainer>
    <Name>{userName}:</Name>
    {children}
  </Container>

export default Message;