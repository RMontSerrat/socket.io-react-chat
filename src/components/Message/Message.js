import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  color: ${props => props.theme.colorPrimary};
  font-family: ${props => props.theme.fontFamily};
  padding: 5px;
`

const DateContainer = styled.span`
  display: block;
  padding-right: 10px;
`
const Name = styled.strong`
  padding-right: 5px;
`

const Message = ({ children, userName, system, date }) =>
  <Container>
    <DateContainer>[{new Date(date).toLocaleTimeString()}]</DateContainer>
    <Name>{system ? 'Servidor' : userName}:</Name>
    {children}
  </Container>

export default Message;