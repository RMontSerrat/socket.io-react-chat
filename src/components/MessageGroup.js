import React from 'react';
import styled from 'styled-components';

const Container = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => props.theme.colorWhite};
  padding: 15px;
  color: ${props => props.theme.colorPrimary};
  > div {
    margin: 10px 0;
  }
`
const MessageGroup = ({ children }) =>
  <Container>
    {children}
  </Container>

export default MessageGroup;