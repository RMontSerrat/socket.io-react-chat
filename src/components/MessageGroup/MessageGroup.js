import styled from 'styled-components';

const MessageGroup = styled.div`
  border-radius: ${props => props.theme.borderRadius};
  background: ${props => props.theme.colorWhite};
  padding: ${props => props.theme.padding};
  color: ${props => props.theme.colorPrimary};
  > div {
    margin: 10px 0;
  }
`
export default MessageGroup;