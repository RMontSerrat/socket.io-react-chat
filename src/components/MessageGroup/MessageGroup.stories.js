import React from 'react';
import { storiesOf } from '@storybook/react';
import MessageGroup from './MessageGroup';
import Message from '../Message';

storiesOf('MessageGroup', module)
  .add('default', () => (
    <div style={{ backgroundColor: '#025a3f', padding: '20px' }}>
      <MessageGroup>
        <Message userName="Servidor" date={new Date()}>Romulo entrou na sala</Message>
        <Message userName="Romulo" date={new Date()}>Fala galera</Message>
        <Message userName="Servidor" date={new Date()}>Zé entrou na sala</Message>
        <Message userName="Zé" date={new Date()}>e aiii.....</Message>
      </MessageGroup>
    </div>
  ));
