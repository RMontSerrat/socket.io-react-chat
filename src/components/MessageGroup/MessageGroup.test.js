import React from "react";
import { render, cleanup } from "react-testing-library";
import MessageGroup from "./MessageGroup";
import Message from '../Message';
import 'jest-styled-components'

afterEach(cleanup);
describe('<MessageGroup />', () => {
  it("should render", () => {
    const { asFragment } = render(
      <MessageGroup>
        <Message userName="Romulo" date={new Date('2019-01-01')}>Fala galera</Message>
      </MessageGroup>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});