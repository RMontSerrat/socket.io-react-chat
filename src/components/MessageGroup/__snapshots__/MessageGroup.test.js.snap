// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageGroup /> should render 1`] = `
<DocumentFragment>
  .c0 > div {
  margin: 10px 0;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 5px;
}

.c2 {
  display: block;
  padding-right: 10px;
}

.c3 {
  padding-right: 5px;
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        [22:00:00]
      </span>
      <strong
        class="c3"
      >
        Romulo:
      </strong>
      Fala galera
    </div>
  </div>
</DocumentFragment>
`;
